core/variant_op.cpp

```diff
@@ -430,10 +430,13 @@ void Variant::evaluate(const Operator &p_op, const Variant &p_a,
 					_RETURN_FAIL;
 				}
 
-				const Dictionary *arr_a = reinterpret_cast<const Dictionary *>(p_a._data._mem);
-				const Dictionary *arr_b = reinterpret_cast<const Dictionary *>(p_b._data._mem);
+				const Dictionary *dict_a = reinterpret_cast<const Dictionary *>(p_a._data._mem);
+				const Dictionary *dict_b = reinterpret_cast<const Dictionary *>(p_b._data._mem);
 
-				_RETURN(*arr_a == *arr_b);
+				if (dict_a->size() != dict_b->size())
+					_RETURN(false);
+
+				_RETURN(dict_a->hash() == dict_b->hash());
```

Comment:
I think it might be better to compare dictionaries similar to how arrays are compared below: element-by-element.

Reply:
Ok, I can do that. I will also sort the keys so the comparison doesn't depend on the insertion order. Is that ok?

Reply:
Why isn't hash comparison enough though? Isn't it more efficient this way if we only care about equality/inequality?

Reply:
Well, calculating the hash also goes through all the entries, so it is the same.

> I will also sort the keys so the comparison doesn't depend on the insertion order. Is that ok?

No need for sorting, as long as the size matches. If `a.size() == b.size()` and for all `x`, `x` from `a`, `b.has(x) && a[x] == b[x]`, then all keys/values in `a` are also present in `b` and vice-versa.

Reply:
Yep, makes sense for the sorting.

I was thinking I could even make it a bit more efficient with a new method `get_entries` in Dictionary to get a list of key/value pairs directly from `a` without accessing `a[x]` for each key, and then use `b.getptr(x)` to check it has the key and get the value for comparison. Or is it too much?

Reply:
I guess you can do that as well, though it seems a bit complicated.

Reply:
Ok, I ended up doing it the optimal way, it wasn't too difficult.